sdk/management/azure/src/main/java/com/azure/management/Azure.java

```diff
@@ -384,51 +369,43 @@ public SdkContext sdkContext() {
 
         @Override
         public Azure withSubscription(String subscriptionId) {
-            return new Azure(restClient, subscriptionId, tenantId, this);
+            profile.withSubscriptionId(subscriptionId);
+            return new Azure(httpPipeline, profile, this);
         }
 
         @Override
-        public Azure withDefaultSubscription() throws CloudException, IOException {
-            if (this.defaultSubscription != null) {
-                return withSubscription(this.defaultSubscription);
-            } else {
-                PagedIterable<Subscription> subs = this.subscriptions().list();
-                if (subs.iterator().hasNext()) {
-                    return withSubscription(subs.iterator().next().subscriptionId());
-                } else {
-                    return withSubscription(null);
-                }
-            }
+        public Azure withDefaultSubscription() throws CloudException {
+            return new Azure(httpPipeline, profile, this);
```

Comment:
Maybe do a throw if `subscriptionId` is null in profile? If user really want to provide a null `subscriptionId`, we would like them to do it explicitly in `withSubscription(String subscriptionId)`.

Also when will `CloudException` happen? It is scheduled to be deleted from core-mgmt.